### ISSUE TITLE
[Feature request] endpoint for proxy chain

### ISSUE THREAD

**Issue by rampageX** [nadoo/glider#196]
It's always face a problem when use the `Circumvention Service Provider`'s servers to browser site like google, youtube：The reCAPTCHA page often appears.

![2020-11-21_211511](https://user-images.githubusercontent.com/994081/99878357-b9668e80-2c3f-11eb-8537-9300996e98bf.png)

Use a proxy build on private VPS as proxy chain's last proxy will avoid this situation. However, we assume that 10 servers are used for load balancing, and the configuration file is like this：

```
forward=proxy1,proxy-vps
forward=proxy2,proxy-vps
forward=proxy3,proxy-vps
...
forward=proxy8,proxy-vps
forward=proxy9,proxy-vps
forward=proxy10,proxy-vps
```

Well, if proxy is `ssr` or `vmess`, the configuration line is very long and hard to read...

How about add a `endpoint` parameter， then the configuration file will like this：

```
forward=proxy1
forward=proxy2
forward=proxy3
...
forward=proxy8
forward=proxy9
forward=proxy10

endpoint=proxy-vps
```

Similarly, in a corporate environment，maybe a `beginpoint` (the first proxy in the proxy chain) can make the configuration file simple too.

**Comment by nadoo** [comment]
Hi @rampageX , it's easy to implement, but I'm afraid it will increase the complexity of configuration and confuse users.

I was also thinking about other similar scenarios like:
```
forward=tls://host1.com,http://user:pass@
forward=tls://host2.com,http://user:pass@
forward=tls://host3.com,http://user:pass@
```
the only difference is the hostname, so it will help if we use a new scheme like:
```
forwardgroup=tls://{HOST},http://user:pass@#HOST=host1.com,host2.com,host3.com
```

in your situation, it will be:
```
forwardgroup={BEGINPOINT},{PROXY},{ENDPOINT}#BEGINPOINT=xxx&ENDPOINT=yyy&PROXY=host1.com,host2.com,host3.com
```

but if the PROXY contains `,`, it will be hard to parse.

let's keep this issue open to see whether we can find a better design to achieve it.
